### PR TITLE
Added video details page and authentication

### PR DESCRIPTION
**Pull request by sanket4120**
Files to review:

- actions (videoActions, userActions)
- context (videosContext, userContext)
- pages (auth, videoDetails)
- reducers (userReducer, videoReducer)

### CODE REVIEW

**Review by partha8 — APPROVED** [review]
Looks good, good to merger.

File: src/components/authRequired/AuthRequired.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { Navigate, Outlet, useLocation } from 'react-router-dom';
+import { useUser } from '../../context/userContext';
+
+const AuthRequired = () => {
+  const location = useLocation();
+  const { authState } = useUser();
+
+  return authState.isAuthenticated ? (
+    <Outlet />
```
Comment (MarkVed17) [inline src/components/authRequired/AuthRequired.js:9]: Ever since I've used Outlet, I think it's very powerful. Nice use case here.

File: src/components/loader/Loader.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import './loader.css';
+
+const Loader = () => {
+  return (
+    <div className='minheight flex align-items-center justify-content-center loader'>
+      <img src='/assets/loader.gif' alt='Loading...' className='contain' />
```
Comment (MarkVed17) [inline src/components/loader/Loader.js:6]: Good to see a loader use case.

File: src/utils/validate.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      errors[fieldName] = error;
+    }
+  }
+  return { isValid, errors };
+};
+
+const validateAddressField = (name, value, formData) => {
+  switch (name) {
+    case 'firstName':
+      return value.length >= 2 && /^[a-zA-Z ]*$/.test(value)
```
Comment (MarkVed17) [inline src/utils/validate.js:73]: regex 💯

**Review by MarkVed17 — APPROVED** [review]
Super clean code. Covers all edge cases IMO. 
Good to proceed!
